core/runtime/binaryen/runtime_manager.cpp

```diff
@@ -0,0 +1,81 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "runtime/binaryen/runtime_manager.hpp"
+
+#include <binaryen/wasm-binary.h>
+
+#include "crypto/hasher/hasher_impl.hpp"
+#include "runtime/binaryen/runtime_external_interface.hpp"
+
+OUTCOME_CPP_DEFINE_CATEGORY(kagome::runtime::binaryen,
+                            RuntimeManager::Error,
+                            e) {
+  using Error = kagome::runtime::binaryen::RuntimeManager::Error;
+  switch (e) {
+    case Error::EMPTY_STATE_CODE:
+      return "Provided state code is empty, calling a function is impossible";
+    case Error::INVALID_STATE_CODE:
+      return "Invalid state code, calling a function is impossible";
+  }
+}
+
+namespace kagome::runtime::binaryen {
+
+  outcome::result<std::tuple<std::shared_ptr<wasm::ModuleInstance>,
+                             std::shared_ptr<WasmMemory>>>
+  RuntimeManager::getRuntimeEvironment() {
```

Comment:
```suggestion
  RuntimeManager::getRuntimeEnvironment() {
```
It's a fun fact that our whole project is actually Polkadot Runtime Environment. But as this thing was renamed to Polkadot Host I think it's okay to keep this name, don't think it will confuse anybody.

Reply:
Thanks. Typo will fixed